platform/lang-api/src/com/intellij/codeInsight/generation/OverrideImplementsAnnotationsFilter.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.intellij.codeInsight.generation;
+
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Allows plugging in annotations processing during override/implement.
+ * <p/>
+ * Parameter annotations would not be copied, if they are not specified in {@link #getAnnotations(PsiFile)}.
+ */
+public interface OverrideImplementsAnnotationsFilter {
+  ExtensionPointName<OverrideImplementsAnnotationsFilter> EP_NAME =
+    ExtensionPointName.create("com.intellij.overrideImplementsAnnotationsFilter");
+
+  /**
+   * Returns annotations which should be copied from a source to an implementation (by default, no annotations are copied).
+   */
+  @Contract(pure = true)
+  default String[] getAnnotations(@NotNull PsiFile file) {
+    return getAnnotations(file.getProject());
+  }
+
+  /**
+   * @deprecated Use {@link #getAnnotations(PsiFile)}
+   */
+  @Deprecated(forRemoval = true)
```

Comment:
Have here only `String[] getAnnotations(@NotNull PsiFile file)` (abstract) and keep the default impl and old deprecated method in java. There is no need to pull them as is to the new API

Reply:
Thank you for the idea!